dessia_common/datatools/learning_models.py

```diff
@@ -0,0 +1,1779 @@
+        return model
+
+    def _instantiate_skl(self):
+        model = self._call_skl_model()
+        model = self._setstate_dessia(model, self.tree_state)
+        return model
+
+    @classmethod
+    def _instantiate_dessia(cls, model, parameters: Dict[str, Any], name: str = ''):
+        kwargs = {'name': name,
+                  'tree_state': cls._getstate_dessia(model),
+                  'n_classes': model.n_classes.tolist(),
+                  'n_features': model.n_features,
+                  'n_outputs': model.n_outputs}
+        return cls(**kwargs)
+
+
+class DecisionTreeRegressor(Model):
+    """
+    Base class for handling scikit-learn `DecisionTreeRegressor`.
+
+    More information: https://scikit-learn.org/stable/modules/tree.html#tree
+
+    :param parameters:
+        Hyperparameters of the used scikit-learn object.
+    :type parameters: dict[str, Any], `optional`
+
+    :param n_outputs_:
+        The number of outputs when fit is performed.
+    :type n_outputs_: int, `optional`, defaults to `None`
+
+    :param tree_:
+        The underlying Tree object.
+        Please refer to https://scikit-learn.org/stable/auto_examples/tree/plot_unveil_tree_structure.html for
+        attributes of Tree object and understanding the decision tree structure for basic usage of these attributes.
+    :type tree_: Tree, `optional`, defaults to `None`
+
+    :param name:
+        Name of `DecisionTreeRegressor`
+    :type name: str, `optional`, defaults to `''`
+    """
+
+    _standalone_in_db = True
+
+    def __init__(self, parameters: Dict[str, Any], n_outputs_: int = None, tree_: Tree = None, name: str = ''):
+        self.n_outputs_ = n_outputs_
+        self.tree_ = tree_
+        Model.__init__(self, parameters=parameters, name=name)
+
+    @classmethod
+    def _skl_class(cls):
+        return tree.DecisionTreeRegressor
+
+    def generic_skl_attributes(self):
+        """ Generic method to set scikit-learn model attributes from self attributes. """
+        model = self._call_skl_model()
+        model.n_outputs_ = self.n_outputs_
+        model.tree_ = self.tree_._instantiate_skl()
+        return model
+
+    @classmethod
+    def generic_dessia_attributes(cls, model, parameters: Dict[str, Any], name: str = ''):
+        """ Generic method to set self attributes from scikit-learn model attributes. """
+        return {'name': name,
+                'tree_': Tree._instantiate_dessia(model.tree_, None),
+                'n_outputs_': model.n_outputs_,
+                'parameters': parameters}
+
+    def _instantiate_skl(self):
+        return self.generic_skl_attributes()
+
+    @classmethod
+    def _instantiate_dessia(cls, model, parameters: Dict[str, Any], name: str = ''):
+        return cls(**cls.generic_dessia_attributes(model, parameters=parameters, name=name))
+
+    @classmethod
+    def _check_criterion(cls, criterion: str):
+        if 'egressor' not in cls.__name__ and criterion == 'squared_error':
```

Comment:
"egressor" check looks unsafe (in the case of a refactor, for ex). Can't we use Subclass stuff ?

Reply:
Done